ZenPacks/zenoss/HttpMonitor/http.py

```diff
@@ -10,107 +10,117 @@
+        self._timeout = timeout
+        self._ssl = ssl
+        self._reactor = reactor
+        self._clientEdnpoint = None
+        self._proxyIp = None
+        self._reqURL = ""
+        self._startTime = None
+        self._follow = follow
+        self._response = dict()
+        self._headers = Headers({b"User-Agent": [b"Zenoss HttpMonitor"]})
+        self._body = ""
+        self._hostnameIp = list()
+
+    def _getIp(self, response):
+        if response:
+            for a in response[0]:
+                if a.payload.TYPE == dns.A:
+                    self._hostnameIp.append(a.payload.dottedQuad())
+        return self.request()
+
+    def makeURL(self):
+        url_data = URI.fromBytes(self._url)
+        if url_data.scheme:
+            args = {
+                "scheme": url_data.scheme,
+                "host": url_data.host,
+                "port": url_data.port,
+                "path": url_data.path,
+            }
+        else:
+            args = {
+                "scheme": "https" if self._ssl else "http",
+                "hostname": self._hostname,
+                "port": self._port,
+                "path": self._url,
+            }
+        self._reqURL = "{scheme}://{hostname}:{port}{path}".format(**args)
+        url_data = URI.fromBytes(self._url)
+        hasScheme = bool(url_data.scheme)
+        hostMatch = url_data.host == self._hostname
+        ipMatch = self._ipAddr in self._hostnameIp
+        if (hasScheme and not hostMatch) or \
+                (not hasScheme and xor(hostMatch, ipMatch)):
+            self._proxyIp = self._ipAddr
 
     def useProxy(self, username, password):
         proxyAuth = base64.encodestring('%s:%s' % (username, password))
         proxy_authHeader = "Basic " + proxyAuth.strip()
-        self.headers.update({'Proxy-Authorization': proxy_authHeader})
+        self._headers.setRawHeaders('Proxy-Authorization', [proxy_authHeader])
 
     def useAuth(self, username, password):
         basicAuth = base64.encodestring("%s:%s" % (username, password))
         authHeader = "Basic " + basicAuth.strip()
-        AuthHeaders = {"Authorization": authHeader}
-        self.headers.update(AuthHeaders)
+        self._headers.setRawHeaders("Authorization", [authHeader])
+
+    def _agent(self):
+        if not self._proxyIp:
+            agent = Agent(self._reactor, connectTimeout=self._timeout)
+            agent = RedirectAgent(agent) if self._follow else agent
+            return agent.request("GET", self._reqURL, self._headers)
+        elif self._proxyIp:
+            endpoint = TCP4ClientEndpoint(reactor=self._reactor, host=self._ipAddr, port=self._port,
+                                          timeout=self._timeout)
+            agent = ProxyAgent(endpoint)
+            agent = RedirectAgent(agent) if self._follow else agent
+            return agent.request("GET", self._reqURL, self._headers)
+        else:
+            raise Exception("Can't choose a connection type")
```

Comment:
This `else:` will never be executed.  The two prior conditions test `_proxyIp` as if it were a boolean value.  There are only two values, so a third option will never be executed.